java-checks-test-sources/src/main/java/checks/EqualsParametersMarkedNonNullCheck.java

```diff
@@ -0,0 +1,53 @@
+package checks;
+
+import javax.annotation.Nonnull;
+
+public class EqualsParametersMarkedNonNullCheck {
+
+  static class A {
+    public boolean equals(@Nonnull Object obj) { // Noncompliant [[sc=27;ec=35;quickfixes=qf1]] {{"equals" method parameters should not be marked "@Nonnull".}}
+      // fix@qf1 {{Remove "@Nonnull"}}
+      // edit@qf1 [[sc=27;ec=36]] {{}}
+      return true;
+    }
+  }
+
+  static class B {
+    public boolean equals(Object obj) { // Compliant
+      return true;
+    }
+  }
+
+  static class C {
+    public boolean equal() {
+      return true;
+    } // Compliant
+
+    public boolean equals(Object a, Object b) { // Compliant
+      return true;
+    }
+  }
+
+  static class D {
+    public boolean equals(C c) { // Compliant
+      return false;
+    }
+  }
+
+  static class E {
+    public boolean equals(@Nonnull C c) { // Compliant
+      return false;
+    }
+  }
+
+  static class F {
```

Comment:
We could add a test to check that equals in a class annotated "@org.eclipse.jdt.annotation.NonNullByDefault" is not reported by the rule as expected.